### PR TITLE
ref(migrations): add migration target enum

### PR DESCRIPTION
**Pull request by dbanda**
A precursor to #3399 . This adds the target enum to specify which nodes to run an op on.

### Before:
The runner would pass in the local flag to let an op know which nodes to run on

### After:
Ops take in the target enum as a parameter. When executing ops already know what nodes to target. By default, the target enum is set to "unset" and will throw an error if the user does not explicitly set it.

### CODE REVIEW

**Comment by lynnagara** [comment]
The `RunSql` operation is missing the target arg.

**Comment by dbanda** [comment]
> The `RunSql` operation is missing the target arg.

Added

File: snuba/migrations/migration.py [hunk 1]
```diff
@@ -157,9 +157,11 @@ def forwards(self, context: Context, dry_run: bool = False) -> None:
 
         for op in ops:
             if op in local_ops:
-                op.execute(local=True)
+                op.target = OperationTarget.LOCAL
```
Comment (MeredithAnya) [inline snuba/migrations/migration.py:160]: This is temporary right now right? until we add this to the migrations themselves?

Comment (dbanda) [reply snuba/migrations/migration.py:160]: yes

File: snuba/migrations/operations.py [hunk 2]
```diff
@@ -9,26 +10,51 @@
...
 
+class OperationTarget(Enum):
+    """
+    Represents the target nodes of an operation.
+    Either local, distributed.
+    """
+
+    LOCAL = "local"
+    DISTRIBUTED = "distributed"
+    UNSET = "unset"  # target is not set. will throw an error if executed
```
Comment (lynnagara) [inline snuba/migrations/operations.py:21]: Mind commenting this as temporary / to be deprecated in code here?

**Review by lynnagara — APPROVED** [review]

**Pull request merged**